tools/configuration.py

```diff
@@ -1,24 +1,40 @@
 import os
 import sys
-import getopt
+import argparse
 
 from tools import get_pipe_output
 
-default_conf = {
-    'max_domains': 10,
-    'max_ext_length': 10,
-    'style': 'gitstats.css',
-    'max_authors': 7,
-    'max_authors_of_months': 6,
-    'authors_top': 5,
-    'commit_begin': '',
-    'commit_end': 'HEAD',
-    'linear_linestats': 1,
-    'project_name': '',
-    'processes': 8,
-    'start_date': '',
-    'output': 'html'
-}
+GITSTAT_VERSION = '0.91'
+
+
+def get_gitstat_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(prog='repo_stat',
+                                     description='Git repository desktop analyzer. Analyze and generate git statistics '
+                                                 'in HTML format, or export into csv files for further analysis.')
+    parser.add_argument('--max_domains', default=10, type=int, help="Display statistic about top active domains")
+    parser.add_argument('--max_ext_length', default=10, type=int, help="Collect statistic from files has shorter "
+                                                                       "extension as max_ext_length")
+    parser.add_argument('--style', default='gitstats.css', type=str, help="css used as generated html result style")
+    parser.add_argument('--max_authors', default=7, type=int, help="Display statistic about top active authors")
+    parser.add_argument('--max_authors_of_months', default=6, type=int, help="Display statistic about top "
+                                                                             "active authors in monthly statistic")
+    parser.add_argument('--authors_top', default=5, type=int,
+                        help="Display statistic about top active authors in total")
+    parser.add_argument('--commit_begin', default="", type=str, help="Analysis starts from this commit date")
```

Comment:
commit_(begin/end) are not used and do not think will be: scipt does analysis for whole history